README.md

```diff
@@ -108,47 +120,94 @@ import org.mozilla.taskcluster.client.queue.*;
 ...
 ```
 ## Temporary credentials
-```java
-import org.mozilla.taskcluster.client.Credentials;
 
-...
-  //The client's access token
-  String accessToken = "...";
-
-  //client id  
-  String clientId = "...";
+You can generate temporary credentials from permanent credentials using the
+java client. This may be useful if you wish to issue credentials to a third
+party. See http://docs.taskcluster.net/auth/temporary-credentials/ for more
+information. Both named and unnamed temporary credentials are supported,
+although named credentials are preferred if you are not sure which type to use.
 
-  String[] scopes = {...};
+### Example
 
-  // The credentials are valid for a maximum of 31 days only
-  Date start = new Date(...);
-  Date expiry = new Date(...);
-
-  //Unnamed Credentials
-  Credentials credentials = Credentials.createTemporaryCredentials(
-    clientId , accessToken , scopes , start , expiry
-  );
+```java
+package org.mozilla.taskcluster;
 
-...
-  //named credentials
-  String issuer = "...";
+import java.util.Date;
 
-  credentials = Credentials.createTemporaryCredentials(
-    clientId, issuer, accessToken , scopes , start , expiry
-  );
+import org.mozilla.taskcluster.client.APICallFailure;
+import org.mozilla.taskcluster.client.CallSummary;
+import org.mozilla.taskcluster.client.Credentials;
+import org.mozilla.taskcluster.client.EmptyPayload;
+import org.mozilla.taskcluster.client.InvalidOptionsException;
+import org.mozilla.taskcluster.client.queue.ListArtifactsResponse;
+import org.mozilla.taskcluster.client.queue.ListArtifactsResponse.Artifacts;
+import org.mozilla.taskcluster.client.queue.Queue;
+
+/**
+ * This simple demo lists the artifacts in run 0 of task U7On2dUVS9KlEgw7LUaCMQ.
+ * It creates permanent credentials from environment variables
+ * TASKCLUSTER_CLIENT_ID and TASKCLUSTER_ACCESS_TOKEN, and then creates
+ * temporary credentials, valid for 24 hours, from these permanent credentials.
+ * It queries the Queue using the temporary credentials, and with limited
+ * authorized scopes.
```

Comment:
You may want to note that this would not normally be done, and is just illustrating the use of credentials.